sql/core/src/test/scala/org/apache/spark/sql/SQLQuerySuite.scala

```diff
@@ -4750,12 +4750,13 @@ class SQLQuerySuite extends QueryTest with SharedSparkSession with AdaptiveSpark
   }
 
   test("SPARK-47939: Explain should work with parameterized queries") {
-    def checkQueryPlan(df: DataFrame, plan: String): Unit = assert(
-      df.collect()
+    def checkQueryPlan(df: DataFrame, plan: String): Unit = {
```

Comment:
Well, this is only for this single test case instead of the whole `SQLQuerySuite`. It would be great if the PR title could describe more precisely.